### PR TITLE
Migration for syntax unification RFC

### PR DESCRIPTION
**Pull request by TimWhiting**
Migration for the syntax unification proposal.

Should be ready for whenever the RFC is implemented.

### CODE REVIEW

File: packages/riverpod_cli/fixtures/unified_syntax/golden/widgets.dart [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      child: Consumer(
+        builder: (context, ref, child) {
+          return Text('${ref.watch(counterProvider)}');
+        },
+      ),
+    );
+  }
+}
+
+class HooksWatch extends ConsumerHookWidget {
```
Comment (rrousselGit) [inline packages/riverpod_cli/fixtures/unified_syntax/golden/widgets.dart:66]: Could you test custom hooks?

```dart
T useMyHook() {
  <...>
 useProvider(myProvider);
}

void useAnotherHook() {
  useMyHook();
}

Widget build(context) {
  useAnotherHook();
}
```

should become:

```dart
T useMyHook(WidgetReference ref) {
  <...>
  ref.watch(myProvider);
}

void useAnotherHook(WidgetReference ref) {
  useMyHook(ref);
}

Widget build(context. WidgetReference ref) {
  useAnotherHook(ref);
}
```

Comment (TimWhiting) [reply packages/riverpod_cli/fixtures/unified_syntax/golden/widgets.dart:66]: Yeah, I will do this, it will take some more work but it should be doable

Comment (rrousselGit) [inline packages/riverpod_cli/fixtures/unified_syntax/golden/widgets.dart:66]: Another thing to test is `HookBuilder`

```dart
HooKBuilder(
  builder: (context) {
    useProvider(provider);
  }
)
```

->

```dart
HookConsumer(
  builder: (context, ref, child) {
    ref.watch(provider);
  }
)
```

**Comment by rrousselGit** [comment]
In the end I went for `HookConsumerWidget` instead of `ConsumerHookWidget`

This is for the builder variant, which is named `HookConsumer` instead of `ConsumerHookBuilder`

**Comment by TimWhiting** [comment]
I think this gives a pretty solid first draft of the migration tool for these changes. I've made the changes requested, and this should be ready for review again as long as it passes CI.

**Pull request closed without merging**